### PR TITLE
Huge update

### PR DESCRIPTION
**Pull request by nfsprodriver**
- Now using cibersheep's svg icon
- Switched to click version from launchpad
- Fixed Network navigation
- Included Dan Chapman's new-listitem branch
- Replaced context menu by swipe actions bars
- Fixed clipboard clearing
- Fixed hidden 'go to' bar
- Make hidden files option savable (with workaround starting app in /home dir)
- Work with multiple files
- New selection bar with icons
- Added new build instructions (finished?)

### CODE REVIEW

**Comment by Flohack74** [comment]
Wow 180 files changed, I can´t review this manually. @mariogrip can we add Jenkins here so that we have ci test result? Then I would say lets install the click and test it a bit...

**Comment by nfsprodriver** [comment]
I 'edited' not really 180 files. Most is because I've rewritten them from the click repo. You may just review the qml and the root scripts.

**Comment by sverzegnassi** [comment]
In absolute terms, it is not much huge either. I'll try to have a look during the following week. The changes to the build system are not new to me, since File Manager uses some tool and configuration that have been originally written for DocViewer

**Comment by Flohack74** [comment]
Just a note: Weblate does regenerate the po´s out of pot automatically now, so in general please do not touch po files. It will result in a merge conflict on the weblate server which needs manual intervention ;)

**Comment by nfsprodriver** [comment]
@sverzegnassi I'll check your hints. @Flohack74 should I revert to repo's current pos?

**Comment by Flohack74** [comment]
If you have change in po, then yes please :) - But there is another thing, File Manager translations are blocked on Weblate so far. Shall I open them again @sverzegnassi ? I think we could work on the current version a bit.

**Comment by sverzegnassi** [comment]
@Flohack74 No problem, feel free to unlock them. I agree with you: at this point, it might be worth to do some change to the current codebase, I'm going to give it a look while I'm at UbuCon@Paris

File: README-NewBuildInstructions.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+Building a click package
```
[the remark this answers is not in the document]
Comment (nfsprodriver) [reply README-NewBuildInstructions.md]: We should have a deeper look it building the app after merging this, I think.

**Comment by nfsprodriver** [comment]
The PR is finished from my side. Ready to be reviewed and merged :)

**Comment by NeoTheThird** [comment]
From what I can see it looks good, but i'd like a final review from @sverzegnassi... ;)

**Comment by nfsprodriver** [comment]
At any questions, please ask me :)

**Comment by sverzegnassi** [comment]
I started to review the code while I was in Paris, but sadly my chroot environment was broken, therefore I was not able to test it on the phone.
I'll have a check tomorrow, as I come back home.

**Comment by nfsprodriver** [comment]
About the hardcode init path: This is just a workaround to make the newly saveable hidden files option be seen as soon as possible you start to use the app. This can be reverted, but this should be fixed. Didn't know about the broken ContentHub, for me the parts with this I tested worked. I'll check this out later. Thanks!

**Comment by sverzegnassi** [comment]
@nfsprodriver 

- Are we talking about filemanager-app being launched in "/home/"?
If so, I can see the issue: if show_hidden_files is true at start-up, hidden files are not anyway visible.
I will have a look and try to fix this; if possible, I would like to remove the usage of U1DB since it makes the app a bit more complex than necessary.

- What about the visibility of ListItemActions in FolderListView.qml?
I see you check whether it's a hidden file in "/home/phablet" or not. The binding looks a bit heavy and redundant, and can potentially reduce the performance of the application.
If I have a look at FolderListPage.qml, e.g. line 98, for the same action in the grid view, I see a more linear binding:
`Action {
    visible: folderListPage.__pathIsWritable
}`
Is there any specific reason why this wasn't work for you?

**Comment by nfsprodriver** [comment]
@sverzegnassi to point 1: Exactly, I had many attempts to get around that issue. I will have some time next week to check this also. To point 2: This is just a security propose so no user deletes important config files. We should discuss this. Subfolders are deletable.

**Comment by sverzegnassi** [comment]
1. Ok, I'm having a look right now, seems there's something wrong in the C++ code of the FolderListModel. I will push a PR to your branch soon.

2. I would restore it to the earlier policies (i.e. just check if path is writable). It's an issue that was already there, and we can't prevent users from gaining full access by typing their password and deleting system files anyway. We could - in the future - just check if full access has been granted, and toggle hidden files visibility according to that.

I'd really like to get this PR merged soon, so we can start working on the UI, which is full of magic numbers and colors. The app was not in a good state - IMHO - so we probably need to make further pull requests before releasing a new version on the OpenStore.

**Comment by nfsprodriver** [comment]
I'll remove the ~/.* policies. We should discuss that later. Can you tell me more about the ContentHub issue?

**Comment by sverzegnassi** [comment]
@nfsprodriver I've found a few issues (tested with PdfViewer):

- PdfViewer (in the OpenStore) requests imports of single files. However, filemanager-app seems not to respect this settings, and allow multiple files selections.

- If the content sharing request has been aborted from PdfViewer, filemanager-app should react and quit the content-hub selection UI.

- If file-manager-app was already launched, and the new selection controls have been previously used, I was unable to see the content-hub controls for picking a file. The app just kept me showing the new selection controls, instead of content-hub controls. However, I'm not anymore able to reproduce it, since now the application just crashes when I select filemanager as a peer.

**Comment by sverzegnassi** [comment]
Ok, let's keep ContentHub out of the PR for now. ContentHub logic is mixed up with other things at the moment, so I prefer to clean up the code a bit first. Let's merge it!
Thanks again, @nfsprodriver !

**Comment by nfsprodriver** [comment]
About the ContentHub: Maybe there are just some little things to fix. Maybe I did some dumb mistakes which changed the basic behaviour :)

**Comment by Flohack74** [comment]
Wow it was merged! :)

**Comment by NeoTheThird** [comment]
Wow, thanks so much guys, this is awesome! Do you want me to publish a new release to the OpenStore?

**Comment by sverzegnassi** [comment]
@NeoTheThird No, not yet. Personally, I'm not anymore sure about releasing an "intermediate" version before we finish rewriting the app.

There are a lot of small things that doesn't work as expected, duplicated code that works differently each from the other, and a huge amount of implicit name resolving which can be potentially dangerous.
I'd like to address these things before releasing anything, then updating the UI to something that looks a bit better would be just a minor work. :)

Reasonably, ETA is end of September. \o/

**Comment by NeoTheThird** [comment]
Ok cool, looking forward to it! :)

**Pull request merged**